First layer cal filament menu (see new PR 1023)

RetireeJay46 opened the pull request:
The purpose of this mod is to allow users to do a first-layer calibration using whatever filament they currently have loaded in the extruder.  As a safety backup, if the hot end has not been preheated, the mod will default to PLA temperatures for the hot end and bed.
(I thought I had excluded the Configuration_Prusa.h file from the commit because it is unchanged from the source code provided for the MK3 printer.)

RetireeJay46 commented:
The only changes important for this pull request are in ultralcd.cpp.  The message to the user is changed (unfortunately, the English version only; I'm not fluent in other languages).  And the functions that implement the first-layer calibration are modified so that if the hot end is already heated, the steps of heating the hot end and bed to PLA temperatures are bypassed.  This allows the user to preheat to any desired filament temperature and then run the first-layer calibration.

PatrikTheDev commented:
@bubnikv @PavelSindler Why didn't you merge this? It is great idea.

zmerlynn commented:
JFYI, this PR has some git-mechanical issues. It looks like you did a merge into your branch and then put up the PR - this won't work. PRs really need to sit "on top" of master. Because of this, there's clutter in the PR itself, e.g. the bang-bang change as well.

You'll know you've succeeded when you can click on the "Commits" tab and see one commit. (In general, unless it's an involved change, most projects prefer squashing to a single commit.)

RetireeJay46 committed:
Changed ultralcd.cpp to allow the user to either select PLA (default)
or another filament to do First Layer Calibration.  The user dialog
now has one more step if the user says PLA is not in the extruder.
The logic of the actual implementation of the first layer calibration
defaults to PLA, but if the hot end is already over 200C then it
simply proceeds with whatever temperature is existing.  It is presumed
that the bed is appropriately heated.

galluslongus commented:
Nice idea. I print mostly with Petg and have to change to Pla when doing the FLC

BrewChef commented:
Would love to see this feature implemented.

RetireeJay46 commented:
I've got a version on my computer with only one commit and only one file changed, hopefully solving the "mechanical issues".  I'm such a Git noob that I can't figure out how to post it as a Pull Request.

RetireeJay46 commented:
OK, thanks to Zach Loafman, this PR is ready for business!

pth3k commented:
Please make this a thing. I am suffering this right now. layer perfect on PLA, use PETG its too high.

Crunch69 commented:
why is this not implemented this not a extra feature this is a necessary function

albegouws commented:
Bumping this as it would be great quality of life feature

3d-gussner commented:
This has been implemented in 3.8.1 can you @RetireeJay46 please close it?

RetireeJay46 commented:
Thanks.
